### ISSUE TITLE
read ENVI error!!!!

### ISSUE THREAD

**Issue by wzxkk** [OSGeo/gdal#8879]
[datas.txt](https://github.com/OSGeo/gdal/files/13510106/datas.txt)
ENVI content

**Comment by jratike80** [comment]
Please give some details. The data sample seems to be plain numbers without any headers or attached header file. Do you have .hdr file somewhere? https://gdal.org/drivers/raster/envi.html

**Comment by wzxkk** [comment]
[data.zip](https://github.com/OSGeo/gdal/files/13519613/data.zip)

**Comment by wzxkk** [comment]
![update1](https://github.com/OSGeo/gdal/assets/22493253/c4f60a99-b096-4232-a110-991ed2ca1f2b)
![update2](https://github.com/OSGeo/gdal/assets/22493253/1ad8e1b7-76a7-4ee9-b046-8ff62bec01ea)
![update3](https://github.com/OSGeo/gdal/assets/22493253/80e5e18c-ea96-4c7c-b3c7-dbf966d86efc)
After changing the content of the data, it is normal, and ENVI files with only 1 and -999 contents will be read incorrectly

**Comment by rouault** [comment]
The dataset is OK. I'm not sure to understand your issue...
```
$ gdalinfo data -checksum
Driver: ENVI/ENVI .hdr Labelled
Files: data
       data.hdr
Size is 304, 232
Coordinate System is:
GEOGCRS["WGS 84",
    DATUM["World Geodetic System 1984",
        ELLIPSOID["WGS 84",6378137,298.257223563,
            LENGTHUNIT["metre",1]]],
    PRIMEM["Greenwich",0,
        ANGLEUNIT["degree",0.0174532925199433]],
    CS[ellipsoidal,2],
        AXIS["latitude",north,
            ORDER[1],
            ANGLEUNIT["degree",0.0174532925199433]],
        AXIS["longitude",east,
            ORDER[2],
            ANGLEUNIT["degree",0.0174532925199433]],
    ID["EPSG",4326]]
Data axis to CRS axis mapping: 2,1
Origin = (108.512496948242003,23.467498779296900)
Pixel Size = (0.002499999944121,-0.002499999944121)
Metadata:
  Band_1=Band 1
Image Structure Metadata:
  INTERLEAVE=BAND
Corner Coordinates:
Upper Left  ( 108.5124969,  23.4674988) (108d30'44.99"E, 23d28' 3.00"N)
Lower Left  ( 108.5124969,  22.8874988) (108d30'44.99"E, 22d53'15.00"N)
Upper Right ( 109.2724969,  23.4674988) (109d16'20.99"E, 23d28' 3.00"N)
Lower Right ( 109.2724969,  22.8874988) (109d16'20.99"E, 22d53'15.00"N)
Center      ( 108.8924969,  23.1774988) (108d53'32.99"E, 23d10'39.00"N)
Band 1 Block=304x1 Type=Int32, ColorInterp=Undefined
  Description = Band 1
  Checksum=33674
```

**Comment by wzxkk** [comment]
Yes, there is no problem reading with gdalinfo, arcgis and envi tools, but there is no problem reading with java, is there a problem binding java libraries?

**Comment by wzxkk** [comment]
Yes, there is no problem in reading with gdalinfo, arcgis and envi tools, but there is an error in reading with java, and only one row and one column can be read. Is there a problem with the library bound to java?

**Comment by jratike80** [comment]
There actually is a problem with gdalinfo. By your screen capture your GDAL version finds another driver just like my GDAL. User must help gdalinfo and tell that the input format is ENVI by adding `-if ENVI`. There must be a way to do the same with Java.
```
gdalinfo data
Driver: NGSGEOID/NOAA NGS Geoid Height Grids
Files: data
Size is 1, 1
```

**Comment by wzxkk** [comment]
thank you very much

**Comment by rouault** [comment]
reopening to investigate the NGSGEOID issue

**Comment by rouault** [comment]
@wzxkk Issue fixed in master and 3.8 branch. Apologies for closing your bug. It turned out that there was a regression in GDAL master (also now fixed) which was hiding the bug to me.
You can try defining the GDAL_SKIP environment variable to NGSGEOID as a workaround
